scipy/cluster/_heap.pxi

```diff
@@ -0,0 +1,73 @@
+# cython: boundscheck=False, wraparound=False, cdivision=True
+import numpy as np
+
+
+cdef class Heap:
```

Comment:
As this seems to be a reduced version of a standard heap, I suppose it may need a more appropriate name and a brief introduction. Besides, some tests for it would be better.